deployer/templates/hawkular-cassandra-node-dynamic-pv.yaml

```diff
@@ -0,0 +1,114 @@
+id: hawkular-cassandra-node-dynamic-pv
+kind: Template
+apiVersion: v1
+name: 'Hawkular Cassandra Node Template: Persistent Storage'
+description: Configures a Cassandra Node to be added to a cluster.
+metadata:
+  name: hawkular-cassandra-node-dynamic-pv
+  labels:
+    metrics-infra: hawkular-cassandra
+parameters:
+- description: Specify prefix for metrics components; e.g. for "openshift/origin-metrics-deployer:v1.1",
+    set prefix "openshift/origin-"
+  name: IMAGE_PREFIX
+  value: openshift/origin-
+- description: Specify version for metrics components; e.g. for "openshift/origin-metrics-deployer:v1.1",
+    set version "v1.1"
+  name: IMAGE_VERSION
+  value: "latest"
+- description: If this Cassandra node should be the 'master' node.
+  name: MASTER
+  value: 'false'
+- description: The persistent volume claim prefix.
+  name: PV_PREFIX
+  value: metrics-cassandra
+- description: The node number for the Cassandra cluster.
+  name: NODE
+  required: true
+- description: The size of storage requested.
+  name: PV_SIZE
+  value: 10Gi
+objects:
+- kind: PersistentVolumeClaim
+  apiVersion: v1
+  metadata:
+    name: "${PV_PREFIX}-${NODE}"
+    labels:
+      metrics-infra: hawkular-cassandra
+    annotations:
+      volume.alpha.kubernetes.io/storage-class: "foo"
```

Comment:
It's the presence of the annotation itself that causes the dynamic provisioning to happen, I used "foo" because that is what our example docs have. I'm not sure if it is possible to specify it without a value or using an empty string.